llvm/test/tools/llvm-symbolizer/approximate-line-handcrafted.s

```diff
@@ -0,0 +1,489 @@
+#sed -i '220c\	.quad 0x1020                            #.Lfunc_begin3 DW_AT_low_pc' main.s
+#sed -i '221c\	.long 0x1026                            #.Lfunc_end3-.Lfunc_begin3 DW_AT_high_pc' main.s
+#sed -i '252c\	.quad 0x30a0                            #.Lfunc_begin0' main.s
+#sed -i '253c\	.quad 0x30a6                            #.Lfunc_end0' main.s
+#sed -i '254c\	.quad 0x1000                            #.Lfunc_begin1' main.s
+#sed -i '255c\	.quad 0x30a6                            #.Lfunc_end3' main.s
+#sed -i '$a\	.long .Lunit_end - .Lunit_start     # unit length\
+#	.Lunit_start:\
+#	.short 4     # version\
+#	.long .Lprologue_end - .Lprologue_start # header length\
+#.Lprologue_start:\
+#	.byte 1                                      # minimum_instruction_length\
+#	.byte 1                                      # maximum_operations_per_instruction\
+#	.byte 1                                      # default_is_stmt\
+#	.byte -5                                     # line_base\
+#	.byte 14                                     # line_range\
+#	.byte 13                                     # opcode_base\
+#	.byte 0, 1, 1, 1, 1, 0, 0, 0, 1, 0, 0, 1     # arguments in standard opcodes\
+#	.byte 0                                      # end of include directories\
+#	.asciz "definitions.h"                       # filename\
+#	.byte 0                                      # reference to dir0\
+#	.byte 0                                      # modification time\
+#	.byte 0                                      # length of file (unavailable)\
+#	.asciz "main.c"                              # filename\
+#	.byte 0                                      # reference to dir0\
+#	.byte 0                                      # modification time\
+#	.byte 0                                      # length of file (unavailable)\
+#	.byte 0                                      # end of filenames\
+#.Lprologue_end:\
+#	.byte 0x05, 66                               # DW_LNS_set_column (80)\
+#	.byte 0x0a                                   # DW_LNS_set_prologue_end\
+#	.byte 0x00, 9, 2                             # DW_LNS_set_address\
+#	.quad 0x30a0                                 # Address Value (0x00000000000030a0)\
+#	.byte 0x01                                   # DW_LNS_copy\
+#	.byte 0x02                                   # DW_LNS_advance_pc\
+#	.uleb128 0x06                                # (addr +=6, op-index +=0)\
+#	.byte 0, 1, 1                                # DW_LNE_end_sequence\
+#	.byte 0x05, 68                               # DW_LNS_set_column (68)\
+#	.byte 0x0a                                   # DW_LNS_set_prologue_end\
+#	.byte 0x00, 9, 2                             # DW_LNE_set_address\
+#	.quad 0x1000                                 # Address Value (0x0000000000001000)\
+#	.byte 0x11                                   # (address += 0,  line += -1,  op-index += 0)\
+#	.byte 0x05, 39                               # DW_LNS_set_column (39)\
```

Comment:
Do you need this many line table ops? Like we don't use is_stmt at all in the symbolizer, so can probably skip any _stmt related operations, yeah? And probably only need a couple of bytes of line table - but this seems to encode about 20 bytes worth of instruction ranges covered by the line table?

Reply:
The line-table section is deciphered from the linked object files. I can work on it's reduction but that requires reduction in the source code. However, I can't delete the is_stmt line operations in the line program(line-table section) , this will cause weird behaviour.

Reply:
Hi @dwblaikie , I tried the approach of deleting some of is_stmt in the line-table entries, but problem is there are lot less is_stmt entries now and some of them we are making it to line-zero(by choice to stress test ```--skip-line-zero``` scenarios with multiple sequences.). Any reduction from here would be a side effect(missing function name from the llvm-symbolizer output).